src/qat/purr/backends/live.py

```diff
@@ -82,8 +82,23 @@ def apply_setup_to_hardware(
         qubit = Qubit(primary_index, resonator, ch1)
         qubit.create_pulse_channel(ChannelType.drive, frequency=5.5e9)
 
+        grid_time = qubit.physical_channel.grid_time
+        if not pulse_hw_x_pi_2_width % grid_time:
+            log.warning(
+                f"Non-hardware-feasible pulse width detected in hardware setup: pulse_hw_x_pi_2_width is not a multiple of the pulse resolution, which is {np.round(grid_time * 1e09, 3)} ns."
```

Comment:
too long a message, say simply "Pulse width must/should be a multiple of {val}"
also why are you logging a warning ? If it's a hard requirement then please raise an error (you could think about rounding up) (depending on semantics)